packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -103,50 +112,122 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
   }
 
   return (
-    <Modal
-      open={true}
-      submit={handleSubmit}
-      close={onCancel}
-      size="max"
-      header={modalTitle}
-      cta="Save"
-      ctaEnabled={saveEnabled}
-      autoFocusSelector="#id-modal-identify-joins-heading"
-    >
-      <div className="my-2 ml-3 mr-3">
-        <div className="row">
-          <div className="col-12">
-            <Field label="Display Name" required {...form.register("name")} />
-            <Field
-              label="Description (optional)"
-              textarea
-              minRows={1}
-              {...form.register("description")}
-            />
-            <Field
-              label="Identifier Type"
-              options={identityTypes.map((i) => i.userIdType)}
-              required
-              {...form.register("userIdType")}
-            />
-            <StringArrayField
-              label="Dimension Columns"
-              value={userEnteredDimensions}
-              onChange={(dimensions) => {
-                form.setValue("dimensions", dimensions);
-              }}
-            />
-            <SQLInputField
-              userEnteredQuery={userEnteredQuery}
-              datasourceId={dataSource.id}
-              form={form}
-              requiredColumns={requiredColumns}
-              identityTypes={identityTypes}
-              queryType="experiment-assignment"
-            />
+    <>
+      {sqlOpen && dataSource && (
+        <EditSqlModal
+          close={() => setSqlOpen(false)}
+          datasourceId={dataSource.id || ""}
+          placeholder={`SELECT\n      ${userEnteredUserIdType}, date\nFROM\n      mytable`}
```

Comment:
This placeholder isn't right.  `date` is not one of the required columns.  We can make a small utility function that takes a list of required columns (and a table name) and returns a placeholder query.

```ts
placeholder={generatePlaceholderQuery(requiredColumns, "viewed_experiment")}
```